scripts/extensions/ai-widget/src/translations/translations-widget.tsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import {ContentDivider, Heading, IconButton, Spacer} from 'superdesk-ui-framework/react';
+import {superdesk} from '../superdesk';
+import TranslationsBody from './translations-body';
+import TranslationFooter from './translations-footer';
+import {ICommonProps, IStateTranslationsTab} from '../ai-assistant';
+import {configuration} from '../configuration';
+
+export default class TranslationsWidget extends React.Component<ICommonProps<IStateTranslationsTab>> {
+    abortController: AbortController;
+
+    constructor(props: ICommonProps<IStateTranslationsTab>) {
+        super(props);
+
+        this.abortController = new AbortController();
+
+        this.generateTranslations = this.generateTranslations.bind(this);
+    }
+
+    generateTranslations() {
+        configuration.translations?.generateTranslations?.(
```

Comment:
This component can't work if `configuration.translations` is not defined - thus you shouldn't be checking it here. Instead, check somewhere before, and pass `configuration.translations` as a prop that would be non-nullable.

Reply:
Yes, I've already put a check when rendering the action on the inital state of the widget, and if it's triggered programatically, I'll see if there are any gaps.